Vulkan-Pipeline-Assistant/mainwindow.cpp

```diff
@@ -212,39 +293,129 @@ QWidget* MainWindow::MakeViewportStateBlock() {
 }
 
 QWidget* MainWindow::MakeRasterizerBlock() {
+    QMap<QString, VkPolygonMode> polygonModes;
+    polygonModes.insert("Fill", VK_POLYGON_MODE_FILL);
+    polygonModes.insert("Line", VK_POLYGON_MODE_LINE);
+    polygonModes.insert("Point", VK_POLYGON_MODE_POINT);
+
     QWidget* container = new QWidget(m_rightTopContainer);
     QLabel* polygonModeLabel = new QLabel("Polygon Mode");
     QComboBox* polygonModeBox = MakeComboBox(container, {"Fill", "Line", "Point"});
+    QObject::connect(polygonModeBox, (void(QComboBox::*)(int))(&QComboBox::currentIndexChanged), [this, polygonModeBox, polygonModes](int index){
+        QString text = polygonModeBox->currentText();
+        PipelineConfig& config = this->m_vulkan->GetConfig();
+        config.writablePipelineConfig.polygonMode = polygonModes.value(text);
+        this->m_vulkan->WritePipelineConfig();
+        this->m_vulkan->Reload(ReloadFlags::EVERYTHING);
+    });
 
     QLabel* rasterizerDiscardLabel = new QLabel("Discard Enable");
     QComboBox* rasterizerDiscardBox = MakeComboBox(container, {"False", "True"});
+    QObject::connect(rasterizerDiscardBox, (void(QComboBox::*)(int))(&QComboBox::currentIndexChanged), [this, rasterizerDiscardBox](int index){
+        QString text = rasterizerDiscardBox->currentText();
+        bool value = text == "True" ? true : false;
+        PipelineConfig& config = this->m_vulkan->GetConfig();
+        config.writablePipelineConfig.rasterizerDiscardEnable = value;
+        this->m_vulkan->WritePipelineConfig();
+        this->m_vulkan->Reload(ReloadFlags::EVERYTHING);
+    });
 
     QLabel* lineWidthLabel = new QLabel("Line Width");
     QLineEdit* lineWidthBox = new QLineEdit("1.0", container);
+    QObject::connect(lineWidthBox, (void(QLineEdit::*)(int))(&QLineEdit::textChanged), [this, lineWidthBox](int index) {
+       QString text = lineWidthBox->text();
+       uint32_t value = uint32_t(std::atoi(text.toStdString().c_str()));
+       PipelineConfig& config = this->m_vulkan->GetConfig();
+       config.writablePipelineConfig.lineWidth = value;
+       this->m_vulkan->WritePipelineConfig();
+       this->m_vulkan->Reload(ReloadFlags::EVERYTHING);
+    });
+
+    //TODO move this maybe?
```

Comment:
What is your thought process around this todo, could it be removed?

Reply:
I'm not sure if it's best to declare all of those "dropdown maps" in one place, or declare each immediately before its corresponding QComboBox as I've done here

Reply:
Make a note for refactoring in trello and remove the todo